bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/api/OpenSprinklerApi.java

```diff
@@ -89,40 +100,153 @@ public abstract void openStation(int station, BigDecimal duration)
      *
-     * @return current draw in milliamperes
-     * @throws CommunicationApiException
-     * @throws
+     * @return current draw in milliamperes or -1 if sensor not supported
+     */
+    public abstract int currentDraw();
+
+    /**
+     * Returns the state of the second sensor.
+     *
+     * @return 1: sensor is active; 0: sensor is inactive; -1: no sensor.
+     */
+    public abstract int getSensor2State();
+
+    /**
+     *
+     * @return The Wifi signal strength in -dB or 0 if not supported by firmware
+     */
+    public abstract int signalStrength();
+
+    /**
+     *
+     * @return The pulses that the flow sensor has given in the last time period, -1 if not supported.
+     */
+    public abstract int flowSensorCount();
+
+    /**
+     * CLOSES all stations turning them all off.
+     *
      */
-    public abstract int currentDraw() throws CommunicationApiException, NoCurrentDrawSensorException;
+    public abstract void resetStations() throws UnauthorizedApiException, CommunicationApiException;
+
+    /**
+     * Returns true if the internal programs are allowed to auto start.
+     *
+     * @return true if enabled
+     */
+    public abstract boolean getIsEnabled();
+
+    public abstract void enablePrograms(Command command) throws UnauthorizedApiException, CommunicationApiException;
 
     /**
      * Returns the water level in %.
      *
      * @return waterLevel in %
-     * @throws CommunicationApiException
-     * @throws
      */
-    public abstract int waterLevel() throws CommunicationApiException;
+    public abstract int waterLevel();
 
     /**
      * Returns the number of total stations that are controllable from the OpenSprinkler
      * device.
      *
      * @return Number of stations as an int.
-     * @throws Exception
      */
-    public abstract int getNumberOfStations() throws Exception;
+    public abstract int getNumberOfStations();
 
     /**
      * Returns the firmware version number.
      *
      * @return The firmware version of the OpenSprinkler device as an int.
      * @throws Exception
      */
-    public abstract int getFirmwareVersion() throws CommunicationApiException;
+    public abstract int getFirmwareVersion() throws CommunicationApiException, UnauthorizedApiException;
+
+    /**
+     * Sends all the GET requests and stores/cache the responses for use by the API to prevent the need for multiple
+     * requests.
+     *
+     * @throws CommunicationApiException
+     * @throws UnauthorizedApiException
+     */
+    public abstract void refresh() throws CommunicationApiException, UnauthorizedApiException;
+
+    /**
+     * Ask the OpenSprinkler for the program names and store these for future use in a List.
+     *
+     * @throws CommunicationApiException
+     * @throws UnauthorizedApiException
+     */
+    public abstract void getProgramData() throws CommunicationApiException, UnauthorizedApiException;
```

Comment:
Yes it is called if a REFRESH command is sent to the channel (OpenSprinklerDeviceHandler.java). Since a user can change the internal programs stored in the device, I thought this was a way for it to be updated without the need to restart the binding.